components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -561,19 +551,24 @@ void JsonRpcService::RemoveChain(const std::string& chain_id,
 
 bool JsonRpcService::SetNetwork(const std::string& chain_id,
                                 mojom::CoinType coin,
+                                const absl::optional<url::Origin>& origin,
                                 bool silent) {
-  auto network_url = GetNetworkURL(prefs_, chain_id, coin);
-  if (!network_url.is_valid()) {
-    return false;
-  }
+  if (!origin) {
+    auto network_url = GetNetworkURL(prefs_, chain_id, coin);
+    if (!network_url.is_valid()) {
+      return false;
+    }
 
-  chain_ids_[coin] = chain_id;
-  network_urls_[coin] = network_url;
-  ScopedDictPrefUpdate update(prefs_, kBraveWalletSelectedNetworks);
-  update->Set(GetPrefKeyForCoinType(coin), chain_id);
+    ScopedDictPrefUpdate update(prefs_, kBraveWalletSelectedNetworks);
+    update->Set(GetPrefKeyForCoinType(coin), chain_id);
+  } else {
+    ScopedDictPrefUpdate update(prefs_, kBraveWalletSelectedNetworksPerOrigin);
+    update->EnsureDict(GetPrefKeyForCoinType(coin))
+        ->Set(origin->Serialize(), chain_id);
+  }
 
   if (!silent) {
-    FireNetworkChanged(coin);
+    FireNetworkChanged(coin, origin);
```

Comment:
This tries to calculcate chain_id but should get if by args instead

Reply:
fixed in 265a042b3edcacbcba64ce56d9bccdc3621bf1e1